python/triton/runtime/jit.py

```diff
@@ -588,17 +588,25 @@ def run(self, *args, grid, warmup, **kwargs):
         bound_args, specialization, options = binder(*args, **kwargs)
 
         # compute cache key
-        key = str(specialization) + str(options)
+        key = tuple(specialization) + tuple(sorted(options.items()))
```

Comment:
NIT: even slightly faster still, you can avoid the concatenation. Only worth ~10 ns but why not.
```suggestion
        key = (tuple(specialization), tuple(sorted(options.items())))
```